server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -235,7 +235,8 @@ protected void doRun() throws Exception {
                         waitForMappingUpdate,
 
                         ActionListener.wrap(v -> executor.execute(this), this::onRejection),
-                        documentParsingProvider
+                        documentParsingProvider,
+                        request.isSimulated()
```

Comment:
What do you mean by "failing the shard"? The only consequence would be that the call to simulate would fail, right (which might even be desirable, because that is what would happen if you were to attempt to actually index data)? Or are there worse consequences that I am not aware of (which is entirely possible since I'm new to this part of the codebase)?

Reply:
I've changed it to not run simulation in a TransportWriteRequest, while attempting to still reuse as much code as possible.